### ISSUE TITLE
Analysis runs - Out of memory / Times out

### ISSUE THREAD

**Issue by luker-GW** [k8-proxy/go-k8s-infra#103]
NCSC testing of glasswall found that small files can lead to our analysis failing, because the documents contain many words and pages and therefore cause our engine to consume large amounts of memory.

<br/>

The corporate filedrop can't complete the analysis on a file because it's either running out of memory or the web app is timing out as teh analysis seems to be taking too long. We should ensure that memory is allowed to grow to a more sensible level like 8GB if required and that the request doesn't time out.

**Comment by GINAGC** [comment]
Sample files added: https://drive.google.com/drive/folders/1J3_qbOVPfkI57SawB9dhIiuyDjfd32_T?usp=sharing

@mohitsengarr, @mahnouman please try one of the sample files and share your initial thoughts on this issue.

**Comment by mohitsengarr** [comment]
@GINAGC I have given a test on these two files.
1. [LargeFile.docx](https://drive.google.com/file/d/1TuyXDbWezW0kW17SuWNNJROxjOiS7AAI/view?usp=sharing)
2. [LargeFile.docm](https://drive.google.com/file/d/1y3sWeHPworsXBxM5x0V8zo1-r1kw5ut2/view?usp=sharing)

I have not encountered the Times Out error neither from `minio-icap` nor `classic` one. However, there are different output from both version.

1. `minio-icap` based CloudSDK is returning `INTERNAL ERROR` for both files.

API response:
```
{
  "errors": [],
  "status": "GW_FAILED",
  "rebuildProcessingStatus": "INTERNAL_ERROR"
}
```
2. `classic` based CloudSDK is able to process the file [LargeFile.docx](https://drive.google.com/file/d/1TuyXDbWezW0kW17SuWNNJROxjOiS7AAI/view?usp=sharing) and returning `GW_FAILED` for the [LargeFile.docm](https://drive.google.com/file/d/1y3sWeHPworsXBxM5x0V8zo1-r1kw5ut2/view?usp=sharing)

API response:
```
{
  "errors": [
    "<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n    <title>Policy Issue</title>\n</head>\n<body>\n    <h1>Document Access Blocked due to Policy</h1>\n    The file does not comply with the current policy\n    <br>\n    The file id blocked is:\n    <strong>8287a28b-051f-49ad-8c9b-7e01b7dcfe9d</strong>\n    <br />\n    <p>\n        For more information contact your system administrator\n        <hr>\n    </p>\n</body>\n</html>"
  ],
  "status": "GW_FAILED"
}
```

Therefore, timeout error is occurred when the request time is elapsed. This request time out is configurable and we can change it accordingly.

**Comment by himanshu2464** [comment]
this is the same issue we did the test, so there are files that need more than 8GB of ram to complete the process but we have an limit the rebuild-pod/processing-pod to use up to 1000Mi of memory. so because of that, it gives an internal error and Gw_rebuild failed error.

**Comment by himanshu2464** [comment]
@GINAGC 
i tried with LargeFile.docx
![image](https://user-images.githubusercontent.com/45463238/127289943-00c6660b-2495-4e75-ad05-e641bdcc3e6a.png)

and it work after changing the memory limit to 10000Mi.

**Comment by GINAGC** [comment]
@himanshu2464 to validate if we can implement this increase in general to 10000Mi and the impact of this.

**Comment by GINAGC** [comment]
@himanshu2464, @mahnouman on this one, the product team suggests setting a limit on memory 8GB and if it fails, send the message that it fails because of the memory. They have also commented that the ultimate fix on the core team.

Can we try this please?

**Comment by himanshu2464** [comment]
@GINAGC I had already try it with 8000Mi and 10000Mi memory limit and it is working correctly, as I have already mentioned that in the above screenshot.

**Comment by GINAGC** [comment]
@himanshu2464 did you also did some tests to confirm that there is not impact if we set this memory as limit? @mahnouman mentioned that more testing was needed, so please confirm.

**Comment by himanshu2464** [comment]
@GINAGC  yes I did the test, and I think there will be no impact because we are not constantly using this limit, but still, I wanted to run the 1 test, and then I will update the deployment script so that it will be present in next build.

**Comment by himanshu2464** [comment]
here is the PR 
https://github.com/k8-proxy/go-k8s-infra/pull/169.
@GINAGC @mahnouman 
kindly review and merge it.

**Comment by GINAGC** [comment]
Moving to QA, we wil have to re run performance test and large files issues.
